sdk/formrecognizer/Azure.AI.FormRecognizer/src/ExtractLabeledFormOperation.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.FormRecognizer.Models
+{
+    /// <summary>
+    /// </summary>
+    internal class ExtractLabeledFormOperation : Operation<IReadOnlyList<ExtractedLabeledForm>>
+    {
+        private Response _response;
+        private IReadOnlyList<ExtractedLabeledForm> _value;
+        private bool _hasCompleted;
+
+        private readonly string _modelId;
+        private readonly ServiceClient _operations;
+
+        public override string Id { get; }
+
+        public override IReadOnlyList<ExtractedLabeledForm> Value => OperationHelpers.GetValue(ref _value);
+
+        public override bool HasCompleted => _hasCompleted;
+
+        public override bool HasValue => _value != null;
+
+        /// <inheritdoc/>
+        public override Response GetRawResponse() => _response;
+
+        /// <inheritdoc/>
+        public override ValueTask<Response<IReadOnlyList<ExtractedLabeledForm>>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(cancellationToken);
+
+        /// <inheritdoc/>
+        public override ValueTask<Response<IReadOnlyList<ExtractedLabeledForm>>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        /// <summary>
+        /// </summary>
+        /// <param name="operations"></param>
+        /// <param name="modelId"></param>
+        /// <param name="operationLocation"></param>
+        internal ExtractLabeledFormOperation(ServiceClient operations, string modelId, string operationLocation)
+        {
+            _operations = operations;
+            _modelId = modelId;
+
+            // TODO: Add validation here
+            // https://github.com/Azure/azure-sdk-for-net/issues/10385
+            Id = operationLocation.Split('/').Last();
+        }
+
+        /// <inheritdoc/>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) =>
+            UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <inheritdoc/>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) =>
+            await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        private async Task<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
```

Comment:
Since this method can be executed 100% synchronously, `ValueTask<Response>` would be better.